cogs/commands/system.py

```diff
@@ -0,0 +1,155 @@
+        await ctx.reply(reply)
+
+    @commands.hybrid_command()
+    async def version(self, ctx: Context[Bot]):
+        """Get Apollo's version"""
+        if self.version_from_file:
+            await ctx.reply(self.version_from_file)
+        else:
+            await ctx.reply("Could not find version")
+
+    @commands.hybrid_command()
+    async def restart(self, ctx: Context[Bot]):
+        headers = {"X-API-Key": f"{CONFIG.PORTAINER_API_KEY}"}
+        async with aiohttp.ClientSession(headers=headers) as session:
+            url = "https://portainer.uwcs.co.uk/api/endpoints/2/docker/containers/apollo/restart"
+            event = SystemEvent(EventKind.RESTART, ctx.channel.id, ctx.message.id)
+            db_session.add(event)
+            db_session.commit()
+            resp = await session.post(url)
+            if not resp.ok:
+                status = resp.status
+                msg = (await resp.json())["message"]
+                logging.error(f"{status} from Portainer API: {msg}")
+                # remove our event that just failed to happen
+                db_session.delete(event)
+                db_session.commit()
+
+    # TODO: same as above but re-create container
+    # have to do it all through Docker API; fetch config, save it, pull image, start contaienr
+
+    @commands.Cog.listener()
+    async def on_ready(self):
+        # check for any unacknowledged events
+        latest, *old = db_session.scalars(
+            select(SystemEvent)
+            .where(SystemEvent.acknowledged == False)
+            .order_by(SystemEvent.time)
+        ).all()
+        if len(old) != 0:
+            logging.warn(
+                "Old unacknowledged system events found, purging old and ackowledging recent"
+            )
+            for event in old:
+                db_session.delete(event)
+
+        channel = await self.bot.fetch_channel(latest.channel_id)
+        if not isinstance(channel, (discord.TextChannel)):
+            logging.error("message id invalid, cannot fetch message id")
+            return
+
+        message = await channel.fetch_message(latest.message_id)
+        name = message.author.display_name
+        await message.reply(
+            f"Hello {name}. Apollo, version {self.version_from_file}, started!"
+        )
+        db_session.commit()
+
+    @staticmethod
+    async def get_docker_json() -> dict[Any, Any] | None:
+        headers = {"X-API-Key": f"{CONFIG.PORTAINER_API_KEY}"}
+        async with aiohttp.ClientSession(headers=headers) as session:
+            url = "https://portainer.uwcs.co.uk/api/endpoints/2/docker/containers/apollo/json"
+            resp = await session.get(url)
+            if not resp.ok:
+                logging.error("Could not reach Portainer API")
+                return None
+            return await resp.json()
+
+
+async def setup(bot: Bot):
+    url = "https://portainer.uwcs.co.uk/api/endpoints/2/docker/containers/apollo/json"
```

Comment:
Put this as a constant var at the top of the file? Easier to change if necessary that way